jetty-core/jetty-util/src/main/java/org/eclipse/jetty/util/resource/PathResource.java

```diff
@@ -256,75 +267,45 @@ public boolean isContainedIn(Resource r)
     @Override
     public URI getTargetURI()
     {
-        if (!targetResolved)
-        {
-            targetPath = resolveTargetPath();
-            targetResolved = true;
-        }
-        if (targetPath == null)
-            return null;
+        checkAlias();
         return targetPath.toUri();
     }
 
-    private Path resolveTargetPath()
+    private void checkAlias()
     {
-        Path abs = path;
-
-        // TODO: is this a valid shortcut?
-        // If the path doesn't exist, then there's no alias to reference
-        if (!Files.exists(path))
-            return null;
-
-        /* Catch situation where the Path class has already normalized
-         * the URI eg. input path "aa./foo.txt"
-         * from an #resolve(String) is normalized away during
-         * the creation of a Path object reference.
-         * If the URI is different from the Path.toUri() then
-         * we will just use the original URI to construct the
-         * alias reference Path.
-         *
-         * We use the method `toUri(Path)` here, instead of
-         * Path.toUri() to ensure that the path contains
-         * a trailing slash if it's a directory, (something
-         * not all FileSystems seem to support)
-         */
-        if (!URIUtil.equalsIgnoreEncodings(uri, toUri(path)))
+        if (targetPath == null)
         {
-            try
-            {
-                // Use normalized path to get past navigational references like "/bar/../foo/test.txt"
-                Path ref = Paths.get(uri.normalize());
-                return ref.toRealPath();
-            }
-            catch (IOException ioe)
+            targetPath = resolveTargetPath();
+            if (Files.exists(targetPath))
```

Comment:
If `targetPath` exists, how can it ever be non-existent?

Reply:
Not sure what you are asking here.

But I don't think we need this check, I will review why I have put it there.

Reply:
Lets say it this way ...

If `targetPath` is not null, because it got returned from `resolveTargetPath()` how can it not pass `Files.exists(targetPath)`?
It would seem this condition is always true.